framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3484,34 +3484,42 @@ NonlinearSystemBase::setupScalingData()
   if (_auto_scaling_initd)
     return;
 
-  const auto & field_variables = _vars[0].fieldVariables();
+  // Want the libMesh count of variables, not MOOSE, e.g. I don't care about array variable counts
+  const auto n_vars = system().n_vars();
 
   if (_scaling_group_variables.empty())
   {
-    _var_to_group_var.reserve(field_variables.size());
-    _num_scaling_groups = field_variables.size();
+    _var_to_group_var.reserve(n_vars);
+    _num_scaling_groups = n_vars;
 
-    for (auto & field_var : field_variables)
-      _var_to_group_var.insert(std::make_pair(field_var->number(), field_var->number()));
```

Comment:
any other variable type beyond field and scalar I would be unaware of?
this code was all to skip scalar variables

Reply:
nopers

Reply:
https://mooseframework.inl.gov/docs/doxygen/moose/classMooseVariableBase.html